test/e2e/handler/broken_configuration_test.go

```diff
@@ -0,0 +1,171 @@
+			"failed running probe 'ping' with after network reconfiguration -> currentState:",
+			"warnings.warn(",
+		}
+	)
+	Context("with invalid field", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateVerificationError:",
+				"desired",
+			}
+		)
+		BeforeEach(func() {
+			By("Creating the policy")
+			err := setDesiredStateWithPolicyAndNodeSelector(DefaultNetworkBroken, createInterfaceWithInvalidField(), map[string]string{"kubernetes.io/hostname": "node01"})
+			if err != nil {
+				return
+			}
+
+			By("Waiting until the node becomes ready again")
+			waitForNodesReady()
+
+			By("Waiting for policy to be ready")
+			waitForDegradedPolicy(DefaultNetworkBroken)
+
+		})
+		AfterEach(func() {
+			By("Remove the policy")
+			deletePolicy(DefaultNetworkBroken)
+
+			By("Reset desired state at all nodes")
+			resetDesiredStateForNodes()
+		})
+		It("should have the policy with fail message", func() {
+
+			for _, failMessage := range failMessages {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+
+			for _, failMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(failMessage))
+			}
+		})
+	})
+
+	Context("with invalid interface name", func() {
+		var (
+			messagesToKeep = [...]string{
+				"libnmstate.error.NmstateLibnmError:",
+				"No suitable device found for this connection",
+			}
+		)
+		BeforeEach(func() {
+			By("Creating the policy")
+			err := setDesiredStateWithPolicyAndNodeSelector(DefaultNetworkBroken, createInterfaceWithInvalidName(), map[string]string{"kubernetes.io/hostname": "node01"})
+			if err != nil {
+				return
+			}
+
+			By("Waiting until the node becomes ready again")
+			waitForNodesReady()
+
+			By("Waiting for policy to be ready")
+			waitForDegradedPolicy(DefaultNetworkBroken)
+
+		})
+		AfterEach(func() {
+			By("Remove the policy")
+			deletePolicy(DefaultNetworkBroken)
+
+			By("Reset desired state at all nodes")
+			resetDesiredStateForNodes()
+		})
+		It("should have the policy with fail message", func() {
+
+			for _, failMessage := range failMessages {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).NotTo(ContainSubstring(failMessage))
+			}
+
+			for _, validMessage := range messagesToKeep {
+				Expect(enactmentConditionsStatus("node01", DefaultNetworkBroken).Find(nmstate.NodeNetworkConfigurationEnactmentConditionFailing).Message).To(ContainSubstring(validMessage))
+			}
+		})
+	})
+
+	Context("with ping fail", func() {
+		var (
+			messagesToKeep = [...]string{
+				"failed to retrieve default gw at runProbes",
+			}
+		)
+		BeforeEach(func() {
```

Comment:
You may want to consider extracting the code in `BeforeEach` to a helper function taking the policy state as a parameter, since all the rest of what `BeforeEach` does is the same